### PR TITLE
Fix issue with summon_creature

### PR DESCRIPTION
**Pull request by Mkalo**
This fixes the issue when summoning non convinceable monsters.

### CODE REVIEW

File: src/luascript.cpp [hunk 1]
```diff
@@ -6726,16 +6726,30 @@ int LuaScriptInterface::luaCreatureGetMaster(lua_State* L)
...
 	Creature* creature = getUserdata<Creature>(L, 1);
 	if (!creature) {
 		lua_pushnil(L);
 		return 1;
 	}
 
 	Creature* master = getCreature(L, 2);
+	bool force = getBoolean(L, 3, false);
 	if (master) {
-		pushBoolean(L, creature->convinceCreature(master));
```
Comment (WibbenZ) [inline src/luascript.cpp]: Have you tested other scripts that use this function to make sure it doesn't break it?
I tested adding the Creature::addSummon(Creature* creature) to Lua (as you pretty much did here) but that still didn't fix the bug.
Tested an older src from nightlies and had the same issue with the old C++ code.

https://github.com/otland/forgottenserver/commit/de73b47e3bd66007584a2017a7481213fc9c22af

**Comment by Mkalo** [comment]
I reverted the commits about this change and there was no bug before. I have tested this PR as well and it fixes the bug.

File: src/luascript.cpp (hunk 1 above)
Comment (Mkalo) [inline src/luascript.cpp]: It won't break it, the default for the force argument is false. It will work exactly the same when force is not set to true.

**Comment by WibbenZ** [comment]
flash

With this update I can still attack summons, both fire elemental and wasps.
And as I said, if I revert the Lua port I can still attack them and that makes me think this is not related to the Lua port nor this function.

@brunominervino could you test this fix aswell to make sure it works for you?

**Comment by Mkalo** [comment]
@WibbenZ That is not the issue. You have understood it wrong. Login in a normal character (not god) and do utevo res "wasp. It will spawn a wasp but wont be able to convince it because setMaster is using convinceCreature internally. It works for gods because they have the flag to convince any creature.

**Comment by WibbenZ** [comment]
Oh okay nvm then

**Comment by Kamenuvol** [comment]
I don't understand what is being forced. It shouldn't be the default behaviour for setMaster?

**Comment by Mkalo** [comment]
@Kamenuvol Well I didn't wanna change it to keep backward compatibility, the thing that is being forced is because it doesn't check if its either summonable or convinceable (probably how the method should've worked from the beginning).

**Comment by comedinha** [comment]
the old code only use addsummon
https://github.com/otland/forgottenserver/commit/de73b47e3bd66007584a2017a7481213fc9c22af#diff-72726bca5c006a50f5f53f0a15423ceaL931
if player have attacking other monster and use utevo res, no have bug?

**Comment by WibbenZ** [comment]
https://github.com/otland/forgottenserver/blob/master/src/creature.cpp#L1124

**Comment by comedinha** [comment]
@WibbenZ the change is on
+				summon->setFollowCreature(nullptr);
 +				summon->setAttackedCreature(nullptr);
 +				summon->updateTargetList();
 +				summon->updateIdleStatus();
but update target list, possibly no have bugs

**Comment by WibbenZ** [comment]
No ide if its called from one of the other ones, you can check.

**Comment by Mkalo** [comment]
The idea of having this is setMaster can be called in another context like if the monster is already attacking a creature.

**Comment by Kamenuvol** [comment]
setMaster should change the master without any checks

**Comment by ranisalt** [comment]
> setMaster should change the master without any checks

I agree with @Kamenuvol here. If a scripter wants to check if the creature is convinceable, just run `creature:isConvinceable()` before.

**Comment by Mkalo** [comment]
Closed due to #2131

**Pull request closed without merging**